### ISSUE TITLE
There is no code to compile value of inst with embedded object property in it 

### ISSUE THREAD

**Issue by KSchopmeyer** [pywbem/pywbem#2277]
The apparent solution to compiling instances that contain impeddedobject properties is to define the value of the property as a string where the string contains the mof for the value of the embedded instance.

The compiler appears not to cover this but drops everything down to cimvalue() to evaluate the property value.  We need to add code to the compiler to test if the property has embedded... qualifier and then get the string value and recall the compiler.   Mike found this issue when trying to compile mocks for SNIA Array profile, CIM_CIMXMLCapabilities.  Have not yet tested but cannot find anywhere in the code that the correct decisions are made, i.e. instance_declaration() where the property in the class has the embedded instance or embedded object qualifier.

NOTE: I do not propose that we ever consider the generality of embeddedobject where the object could be a class and just consider the case of compiling an instance that contains an embedded instance property.

See DSP004 Appendix F1 for the rules and a example.

```
instance of CIM_CIMXMLCapabilities as $CIMXMLCap {
         InstanceID = "CIMXMLCapabilities Instance";
         AuthenticationMechanismsSupported = {0};
         ElementName = "WBEM Server CIMXMLCapabilities";
         GenericOperationCapabilities = "instance of CIM_GenericOperationCapabilitiesStructure {FQLSupported = false; CQLSupport = {0}; Validated = false; };";
         MaxConnections = 1;
         MultipleOperationsSupported = false;
         RequestedStatesSupported = {2,3,11};
         Validated = false;
         WBEMProtocolVersionsSupported = {2.0.0}; };

**Comment by FarmerMike252** [comment]
I was able to successfully create a class with an embedded structure. Here is the run that enumerates the class:
C:\Users\FarmerMike>workon pipshrub
(pipshrub) C:\Users\FarmerMike\devenv>pywbemcli --name ArrayMock
Enter 'help' for help, <CTRL-D> or ':q' to exit pywbemcli.
pywbemcli> instance enumerate CIM_CIMXMLCapabilities
-Loading classes into the Mock Repository
-Loading instances into the Mock Repository
|DONE Loading instances into the Mock Repository
instance of CIM_CIMXMLCapabilities {
   InstanceID = "CIMXMLCapabilities Instance";
   AuthenticationMechanismsSupported = { 0 };
   ElementName = "WBEM Server CIMXMLCapabilities";
   GenericOperationCapabilities =
      "instance of CIM_GenericOperationCapabilitiesStructure {FQLSupported = "
      "false; CQLSupport = {0}; Validated = false; };";
   MaxConnections = 1;
   MultipleOperationsSupported = false;
   RequestedStatesSupported = { 2, 3, 11 };
   Validated = false;
   WBEMProtocolVersionsSupported = { "2.0.0" };
};

**Comment by KSchopmeyer** [comment]
Note that this displays the mof for the instance as a string.  To be honest, I am not sure exactly what the form should be here.

**Comment by KSchopmeyer** [comment]
Given the following classes:

```

    class CIM_Embedded {
        boolean FQLSupported;
        boolean CQLSupport;
        boolean Validated;
    };

         [Description ("blah blah")]
    class CIM_Foo {
            [Key, Description ("This is key prop")]
        string InstanceID;
            [Description ( "Allocation state of the managed element." ),
                EmbeddedInstance ( "CIM_Embedded" )]
        string Embedded1;
    };

And an instance mof of the following form:

```
instance_mof = """
    instance of CIM_Foo {
        InstanceID = "test1";
        Embedded1 =
            "instance of CIM_Embedded { FQLSupported = True; };";
    };


This is what a mof display of that instance looks like:

```

//  Path=/root/cimv2:CIM_Foo.InstanceId="blah2" 
instance of CIM_Foo {
   InstanceId = "blah2";
   Embedded1 = "instance of Embedded {\n   FQLSupported = true;\n};\n";
};

TODO: Show what the xml and repr look like.

**Comment by KSchopmeyer** [comment]
I concluded that there is no code in the MOF compilerto compile the value of an instance property that has embeddedinstance or embeddedobject qualifier to a CIMInstance that could be used internally.  It just keeps the string.

**Comment by andy-maier** [comment]
Addressed in PR #2331

**Comment by andy-maier** [comment]
@FarmerMike252 Just for clarification: The representation of an embedded instance / class follows from the representation of the containing object:
* If containing object is represented as MOF: Embedded object is represented as a CIM string value containing the MOF string
* If containing object is represented as CIM-XML: Embedded object is represented as CIM-XML INSTANCE/CLASS element (i.e. not as MOF string)
* If containing object is represented as pywbem CIM object (e.g. CIMInstance): Embedded object is represented as pywbem CIM object (i.e. not as MOF string).

This is why Karl said embedded instances were not supported (that way) in the MOF compiler.
